chromium_src/content/public/common/url_constants.cc

```diff
@@ -1,10 +1,12 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "../../../../../content/public/common/url_constants.cc"
 
 namespace content {
 const char kBraveUIScheme[] = "brave";
+const char kBinanceScheme[] = "com.brave.binance";
```

Comment:
any particular reason we don't use `brave://` for this? (which has been somewhat tested to have the security protections of chrome://)

Reply:
we probably should add this scheme to https://github.com/brave/brave-core/blob/1a3d1b3ba07925f8080b7f107e123d8e294e42b8/common/brave_content_client.cc

Reply:
Could consider switching but I believe at this point Binance already has that callback URL and timing is really close to release.  Code changes would also be quite a bit bigger. We're doing the same as Uphold oauth right now but with a bunch of extra protections.  Mainly right now it works as an external protocol handler.   I'm not sure if changing would imply that we need to create a webui page for it or something like that.  I'll reconsider for the next protocol handler that we need to add if that is ok. And at that point possibly suggest changing the uphold one, the binance one, and the new one.

Reply:
we may also want to add it in these files since according to git history, they were all security fixes for brave://: https://github.com/brave/brave-core/blob/2aafcf763803700f148d7328cdac7f653dd0a604/patches/chrome-browser-extensions-extension_tab_util.cc.patch, https://github.com/brave/brave-core/blob/2aafcf763803700f148d7328cdac7f653dd0a604/patches/chrome-browser-profiles-profile_io_data.cc.patch, https://github.com/brave/brave-core/blob/2aafcf763803700f148d7328cdac7f653dd0a604/patches/content-browser-child_process_security_policy_impl.cc.patch, https://github.com/brave/brave-core/blob/2aafcf763803700f148d7328cdac7f653dd0a604/patches/content-renderer-render_thread_impl.cc.patch

also https://github.com/brave/brave-core/blob/96ccf2169e98b54410cc1f842452ffc38785ecad/chromium_src/chrome/browser/history/history_utils.cc if you don't want it to be part of the history

there are also some others in https://github.com/brave/brave-core/search?p=1&q=kBraveUIScheme&unscoped_q=kBraveUIScheme; please double check that none of those are necessary for kBinanceScheme

Reply:
I think https://github.com/brave/brave-core/blob/2aafcf763803700f148d7328cdac7f653dd0a604/patches/chrome-browser-profiles-profile_io_data.cc.patch makes it more allowed than it is currently

Reply:
ditto https://github.com/brave/brave-core/blob/2aafcf763803700f148d7328cdac7f653dd0a604/patches/content-browser-child_process_security_policy_impl.cc.patch

Reply:
https://github.com/brave/brave-core/blob/2aafcf763803700f148d7328cdac7f653dd0a604/patches/content-renderer-render_thread_impl.cc.patch is already handled in this pull request

Reply:
@diracdeltas I'm not following your reasoning for this, sites are not allowed to redirect to `chrome://` or `brave://` urls so using `brave://` here would require us to bypass those security restrictions. What is the reason why we would want to use `brave://` here? What are we trying to protect against?

Reply:
@bridiver it wasn't clear to me that this scheme was needed for external sites to redirect to

Reply:
i think ideally you want an implementation that has *all* the security protections already added to chrome:// or brave:// except for one exception. that seems easier to guarantee if you start with chrome:// or brave:// and remove that one exception rather than start with a new protocol and add the security protections one by one. as we've seen in the past with brave://, there are multiple security protections that we weren't aware of with chrome:// that we missed.

Reply:
for instance this could be done using brave:// by allowing a security exception for brave://binance only

Reply:
I think it's more dangerous to use brave:// because of most of the links above make it more allowed and in different cases than what we need for com.brave.binance.  I pushed out a fix by the way for the originator check . The uphold one that is in use today is wrong by the way, I think in a follow up it should be refactored to use the same code
https://github.com/brave/brave-core/blob/master/components/brave_rewards/browser/rewards_protocol_handler.cc#L62

Reply:
so if most of those patches above are mostly adding security allowances instead of restrictions, can we make a master list of what security restrictions need to be added whenever a new scheme is added to make it as restricted as possible? (and then exceptions can be added one by one)

Reply:
@diracdeltas I created some documentation here as a starting point: https://github.com/brave/brave-browser/wiki/Adding-a-protocol-scheme-to-Brave
Feel free please to fill in extra or anything you think is missing.